tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter.cc

```diff
@@ -561,31 +590,47 @@ StatusOr<bool> F8GraphConv(HloComputation* comp, se::CudaComputeCapability cc) {
         continue;
       }
 
-      std::vector<HloInstruction*> operands;
+      std::vector<HloInstruction*> operands, aux_outputs;
       GraphString graph_string;
       HloInstruction* final_instr;
-      std::tie(operands, graph_string, final_instr) = CaptureConvGraph(
-          const_cast<HloInstruction*>(instr), x_scale, w_scale,
-          x_scale_op ? x_scale_op->opcode() == HloOpcode::kMultiply : false,
-          w_scale_op ? w_scale_op->opcode() == HloOpcode::kMultiply : false);
+
+      TF_ASSIGN_OR_RETURN(
+          std::tie(operands, aux_outputs, graph_string, final_instr),
+          CaptureConvGraph(
+              instr, convolution, wide_input, wide_filter, x_scale, w_scale,
+              x_scale_op ? x_scale_op->opcode() == HloOpcode::kMultiply : false,
+              w_scale_op ? w_scale_op->opcode() == HloOpcode::kMultiply
+                         : false));
       TF_ASSIGN_OR_RETURN(
           auto config, convolution->backend_config<CudnnConvBackendConfig>());
       config.set_serialized_graph(graph_string.Graph());
       operands.insert(operands.begin(), input);
       operands.insert(operands.begin() + 1, filter);
 
-      Shape new_shape = ShapeUtil::MakeTupleShape(
-          {ShapeUtil::ChangeElementType(
-               ShapeUtil::GetTupleElementShape(convolution->shape(), 0),
-               final_instr->shape().element_type()),
-           ShapeUtil::GetTupleElementShape(convolution->shape(), 1)});
-      HloInstruction* new_convolution = comp->AddInstruction(
-          convolution->CloneWithNewOperands(new_shape, operands));
+      std::vector<Shape> output_shapes = {
+          ShapeUtil::ChangeElementType(
+              ShapeUtil::GetTupleElementShape(convolution->shape(), 0),
+              final_instr->shape().element_type()),
+          ShapeUtil::GetTupleElementShape(convolution->shape(), 1)};
+      for (HloInstruction* aux_output : aux_outputs) {
+        output_shapes.insert(output_shapes.begin() + 1, aux_output->shape());
+      }
```

Comment:
Instead of inserting the conv output and the workspace buffer, then inserting each aux output in between them, I would insert the conv output, then the aux outputs, then the workspace buffer.